sound/soc/sof/ipc4-pcm.c

```diff
@@ -189,13 +291,14 @@ static int sof_ipc4_pcm_trigger(struct snd_soc_component *component,
 	}
 
 	/* set the pipeline state */
-	return sof_ipc4_trigger_pipelines(component, substream, state);
+	return sof_ipc4_trigger_pipelines(component, substream, state, cmd);
 }
 
 static int sof_ipc4_pcm_hw_free(struct snd_soc_component *component,
 				struct snd_pcm_substream *substream)
 {
-	return sof_ipc4_trigger_pipelines(component, substream, SOF_IPC4_PIPE_RESET);
+	/* command is not relevant with RESET, so just pass 0 */
```

Comment:
but nothing is done for cmd=0, so why bother?

Reply:
not getting it. why bother as in? why bother with the comment or why bother passing 0?

Reply:
why bother calling a function with a parameter zero which is not handled in the function?

Reply:
because the function needs the parameter? It is called during trigger not just hw_free

Reply:
0 = SNDRV_PCM_TRIGGER_STOP，can we use it ?

Reply:
@RanderWang its not the command itself but the fact that when setting the RESET state that the command doesnt matter

Reply:
yes, I understand it is not used and like some padding bit.  The comments "so just pass 0" make me uncomfortable